avogadro/qtplugins/quantumoutput/quantumoutput.cpp

```diff
@@ -221,13 +241,24 @@ void QuantumOutput::calculateElectronDensity(float isoValue, float stepSize)
   calculateMolecularOrbital(-1, isoValue, stepSize);
 }
 
+void QuantumOutput::displayCube(int cubeIndex, float isoValue)
+{
+  m_cube = m_cubes[cubeIndex - 1];
+  m_isoValue = isoValue;
+  calculateFinished();
+}
+
 void QuantumOutput::calculateFinished()
 {
   qDebug() << "The calculation finished!";
   if (!m_cube)
     return;
 
-  disconnect(&m_concurrent->watcher(), 0, 0, 0);
+  // Barry: I have no clue what disconnect does, but if statement
```

Comment:
Don't need name, disconnect removes all signal-slot connections to the watcher. Your changes mean there is not always a valid watcher as no QtConcurrent job is initiated. I can take a pass and just clean up some of the comments/formatting as this is pretty close. You can't disconnect from a NULL, the debug statement is less valid now too, and we should probably just refactor the method for calculation and display as the two can now be entirely distinct.